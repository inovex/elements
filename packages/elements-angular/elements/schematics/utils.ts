import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
