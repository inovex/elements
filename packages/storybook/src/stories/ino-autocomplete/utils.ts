import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
