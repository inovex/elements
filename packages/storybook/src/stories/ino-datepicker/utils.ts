import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
