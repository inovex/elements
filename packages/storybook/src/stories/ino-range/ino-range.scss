.story-range {
  .flex-parent {
    display: flex;
    justify-content: space-evenly;
    flex-wrap: wrap;

    > .flex-child {    
      display: flex;
      flex-direction: column;
      align-items: center;
      margin: 0 20px;
    }
  }
}
