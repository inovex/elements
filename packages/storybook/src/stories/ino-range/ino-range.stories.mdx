import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-range/readme.md";

<Meta title="Input/ino-range/Documentation" />

<Description markdown={readme} />
