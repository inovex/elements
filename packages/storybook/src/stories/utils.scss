/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
