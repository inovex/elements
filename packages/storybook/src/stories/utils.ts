import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
