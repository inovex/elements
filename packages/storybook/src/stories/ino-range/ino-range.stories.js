import { storiesOf } from '@storybook/html';

import { number, boolean, select, text } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-range/readme.md';
import './ino-range.scss';



storiesOf('Input|<ino-range>', module)
  .addDecorator(withStencilReadme(componentReadme))
  .add('Default usage', () => /*html*/`
    <div class="story-range">
      <div class="flex-parent-center">
        <div>
          <h4>Customizable Range</h4>
          <ino-range
            class="customizable-range"
            autofocus="${boolean('autofocus', false)}"
            disabled="${boolean('disabled', false)}"
            min="${number('min', 0)}"
            max="${number('max', 100)}"
            name="${text('name', '')}"
            ino-discrete="${boolean('ino-discrete', false)}"
            ino-markers="${boolean('ino-markers', false)}"
            value="${number('value', 50)}"
            required="${boolean('required', false)}"
            step="${number('step', 1)}"
            ino-color-scheme="${select(
              'ino-color-scheme',
              ['primary', 'secondary', 'success', 'warning',
              'error', 'light', 'dark'
              ],
              'primary'
            )}">
          </ino-range>
        </div>
      </div>

      <h4>Different Colors</h4>

      <div class="flex-parent">
        <div class="flex-child">
          <h5>primary</h5>
          <ino-range ino-color-scheme="primary"></ino-range>
        </div>
        <div class="flex-child">
          <h5>secondary</h5>
          <ino-range ino-color-scheme="secondary"></ino-range>
        </div>
        <div class="flex-child">
          <h5>success</h5>
          <ino-range ino-color-scheme="success"></ino-range>
        </div>
        <div class="flex-child">
          <h5>warning</h5>
          <ino-range ino-color-scheme="warning"></ino-range>
        </div>
        <div class="flex-child">
          <h5>error</h5>
          <ino-range ino-color-scheme="error"></ino-range>
        </div>
        <div class="flex-child">
          <h5>light</h5>
          <ino-range ino-color-scheme="light"></ino-range>
        </div>
        <div class="flex-child">
          <h5>dark</h5>
          <ino-range ino-color-scheme="dark"></ino-range>
        </div>
      </div>

      <h4>Stepped Range</h4>

      <div class="flex-parent stepped-range-container">
        <div class="flex-child">
          <ino-range ino-color-scheme="primary" min="0" max="100" step="20" ino-discrete ino-markers></ino-range>
        </div>
      </div>
    </div>
  `);
