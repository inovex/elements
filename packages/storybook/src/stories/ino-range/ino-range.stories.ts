import { Components } from '@inovex.de/elements';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator, withColorScheme } from '../utils';
import './ino-range.scss';

export default {
  title: 'Input/<ino-range>',
  component: 'ino-range',
  decorators: [story => defaultDecorator(story, 'story-range')],
};

export const Playground: Story<Components.InoRange> = args => html`
  <ino-range
    class="customizable-range"
    disabled="${args.disabled}"
    min="${args.min}"
    max="${args.max}"
    name="${args.name}"
    discrete="${args.discrete}"
    markers="${args.markers}"
    value="${args.value}"
    step="${args.step}"
    color-scheme="${args.colorScheme}">
  </ino-range>
`;

Playground.args = {
  disabled: false,
  discrete: false,
  min: 0,
  max: 100,
  name: '',
  markers: false,
  value: 50,
  step: 1,
};
withColorScheme(Playground, 'colorScheme');

export const Colors = () => html`
  <div class="flex-parent">
    <div class="flex-child">
      <h5>primary</h5>
      <ino-range color-schem="primary" min="0"  max="100" value="50"></ino-range>
    </div>
    <div class="flex-child">
      <h5>secondary</h5>
      <ino-range color-scheme="secondary" min="0"  max="100" value="50"></ino-range>
    </div>
    <div class="flex-child">
      <h5>success</h5>
      <ino-range color-scheme="success" min="0"  max="100" value="50"></ino-range>
    </div>
    <div class="flex-child">
      <h5>warning</h5>
      <ino-range color-scheme="warning" min="0"  max="100" value="50"></ino-range>
    </div>
    <div class="flex-child">
      <h5>error</h5>
      <ino-range color-scheme="error" min="0"  max="100" value="50"></ino-range>
    </div>
    <div class="flex-child">
      <h5>light</h5>
      <ino-range color-scheme="light" min="0"  max="100" value="50"></ino-range>
    </div>
    <div class="flex-child">
      <h5>dark</h5>
      <ino-range color-scheme="dark" min="0"  max="100" value="50"></ino-range>
    </div>
  </div>
`;

export const SteppedRange = () => html`
  <ino-range color-scheme="primary" min="0"  max="100" value="50" step="20" discrete markers></ino-range>
`;
