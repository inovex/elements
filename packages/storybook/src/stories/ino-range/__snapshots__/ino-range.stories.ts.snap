// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input/<ino-range> Discrete smoke-test 1`] = `
<div id="root-inner">
  <div class="story-range">
    <ino-range class="customizable-range ino-range--color-scheme-undefined hydrated"
               disabled="false"
               min="0"
               max="100"
               name
               discrete="true"
               markers="false"
               value="50"
               step="1"
               ranged="false"
               value-start="30"
               value-end="70"
               color-scheme="undefined"
    >
      <div class="mdc-slider mdc-slider--discrete">
        <input class="mdc-slider__input"
               type="range"
               min="0"
               max="100"
               step="1"
               name
               aria-label
               value="70"
        >
        <div class="mdc-slider__track">
          <div class="mdc-slider__track--inactive">
          </div>
          <div class="mdc-slider__track--active">
            <div class="mdc-slider__track--active_fill"
                 style="transform: scaleX(0.7);"
            >
            </div>
          </div>
        </div>
        <div class="mdc-slider__thumb mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
             style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px; transform: translateX(840px);"
        >
          <div class="mdc-slider__value-indicator-container"
               aria-hidden
          >
            <div class="mdc-slider__value-indicator">
              <span class="mdc-slider__value-indicator-text">
                70
              </span>
            </div>
          </div>
          <div class="mdc-slider__thumb-knob">
          </div>
        </div>
      </div>
    </ino-range>
  </div>
</div>
`;

exports[`Input/<ino-range> Markers smoke-test 1`] = `
<div id="root-inner">
  <div class="story-range">
    <ino-range class="customizable-range ino-range--color-scheme-undefined hydrated"
               disabled="false"
               min="0"
               max="100"
               name
               discrete="true"
               markers="true"
               value="50"
               step="1"
               ranged="false"
               value-start="30"
               value-end="70"
               color-scheme="undefined"
    >
      <div class="mdc-slider mdc-slider--discrete mdc-slider--tick-marks">
        <input class="mdc-slider__input"
               type="range"
               min="0"
               max="100"
               step="1"
               name
               aria-label
               value="70"
        >
        <div class="mdc-slider__track">
          <div class="mdc-slider__track--inactive">
          </div>
          <div class="mdc-slider__track--active">
            <div class="mdc-slider__track--active_fill"
                 style="transform: scaleX(0.7); transition: all 0s ease 0s;"
            >
            </div>
          </div>
          <div class="mdc-slider__tick-marks">
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--active">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
            <div class="mdc-slider__tick-mark--inactive">
            </div>
          </div>
        </div>
        <div class="mdc-slider__thumb mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
             style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px; transform: translateX(840px); transition: all 0s ease 0s;"
        >
          <div class="mdc-slider__value-indicator-container"
               aria-hidden
          >
            <div class="mdc-slider__value-indicator">
              <span class="mdc-slider__value-indicator-text">
                70
              </span>
            </div>
          </div>
          <div class="mdc-slider__thumb-knob">
          </div>
        </div>
      </div>
    </ino-range>
  </div>
</div>
`;

exports[`Input/<ino-range> Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-range">
    <ino-range class="customizable-range"
               disabled="false"
               min="0"
               max="100"
               name
               discrete="false"
               markers="false"
               value="50"
               step="1"
               ranged="false"
               value-start="30"
               value-end="70"
               color-scheme="undefined"
    >
    </ino-range>
  </div>
</div>
`;

exports[`Input/<ino-range> Ranged smoke-test 1`] = `
<div id="root-inner">
  <div class="story-range">
    <ino-range class="customizable-range ino-range--color-scheme-undefined hydrated"
               disabled="false"
               min="0"
               max="100"
               name
               discrete="true"
               markers="false"
               value="50"
               step="1"
               ranged="true"
               value-start="30"
               value-end="70"
               color-scheme="undefined"
    >
      <div class="mdc-slider mdc-slider--discrete mdc-slider--range">
        <input class="mdc-slider__input"
               type="range"
               min="0"
               max="70"
               value="30"
        >
        <input class="mdc-slider__input"
               type="range"
               min="30"
               max="100"
               step="1"
               name
               aria-label
               value="70"
        >
        <div class="mdc-slider__track">
          <div class="mdc-slider__track--inactive">
          </div>
          <div class="mdc-slider__track--active">
            <div class="mdc-slider__track--active_fill"
                 style="transform-origin: left center; right: unset; left: 360px; transform: scaleX(0.4); transition: all 0s ease 0s;"
            >
            </div>
          </div>
        </div>
        <div class="mdc-slider__thumb mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
             style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px; transform: translateX(360px); transition: all 0s ease 0s;"
        >
          <div class="mdc-slider__value-indicator-container"
               aria-hidden
          >
            <div class="mdc-slider__value-indicator">
              <span class="mdc-slider__value-indicator-text">
                30
              </span>
            </div>
          </div>
          <div class="mdc-slider__thumb-knob">
          </div>
        </div>
        <div class="mdc-slider__thumb mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
             style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px; transform: translateX(840px); transition: all 0s ease 0s;"
        >
          <div class="mdc-slider__value-indicator-container"
               aria-hidden
          >
            <div class="mdc-slider__value-indicator">
              <span class="mdc-slider__value-indicator-text">
                70
              </span>
            </div>
          </div>
          <div class="mdc-slider__thumb-knob">
          </div>
        </div>
      </div>
    </ino-range>
  </div>
</div>
`;
